src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -456,6 +821,100 @@ void CodegenNeuronCppVisitor::print_global_variables_for_hoc() {
     printer->add_line("{nullptr, nullptr, 0}");
     printer->decrease_indent();
     printer->add_line("};");
+
+    printer->add_newline(2);
+    printer->add_line("/* declaration of user functions */");
+    for (const auto& procedure: info.procedures) {
+        const auto proc_name = procedure->get_node_name();
+        if (proc_name[0] == '_') {
```

Comment:
what this means? i.e. NMODL doesn't allow methods/variables with `_`? and if its allowed, why they need to be skipped declarations?

Reply:
This is case is related to https://github.com/neuronsimulator/nrn/blob/d7750508d713877bd55e25186fa3b56df615ae16/src/nmodl/nocpout.cpp#L222. I think there is some special handling if the `SUFFIX` of a mechanism is set to `nothing`.